src/safeposix/syscalls/sys_calls.rs

```diff
@@ -236,29 +291,32 @@ impl Cage {
             let mainsigset = interface::RustAtomicU64::new(
                 mainsigsetatomic.load(interface::RustAtomicOrdering::Relaxed),
             );
+            // Insert the parent cage object's main threads sigset and store them at index 0
             newsigset.insert(0, mainsigset);
         }
 
-        /*
-         *  Construct a new semaphore table in child cage which equals to the one in the parent cage
-         */
+        // Construct a new semaphore table in child cage which equals to the one in the parent cage
+        // 
         let semtable = &self.sem_table;
         let new_semtable: interface::RustHashMap<
             u32,
             interface::RustRfc<interface::RustSemaphore>,
         > = interface::RustHashMap::new();
-        // Loop all pairs
+        // Loop all pairs of semaphores and insert their copies into the new semaphore table
         for pair in semtable.iter() {
```

Comment:
Is there any example we can show here about how the semaphore pair looks like, to understand easily.

Reply:
Added!